tests/bsim/bluetooth/host/misc/sample_test/src/main.c

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+
+#include "bs_tracing.h"
+#include "bstests.h"
+
+extern void entrypoint_dut(void);
+extern void entrypoint_peer(void);
+extern enum bst_result_t bst_result;
+
+
+static void test_tick(bs_time_t HW_device_time)
+{
+	/* This callback will fire right before the simulation times out.
+	 *
+	 * You can use it to print test or system state that would be of use for
+	 * debugging why the test timed out.
+	 *
+	 * Can also be used to trigger a `k_oops` which will halt the image if
+	 * running under a debugger, if `CONFIG_ARCH_POSIX_TRAP_ON_FATAL=y`.
+	 */
+
+	if (bst_result != Passed) {
+		bs_trace_error_time_line("Simulation ends now.\n");
+	}
+}
+
+static const struct bst_test_instance entrypoints[] = {
+	{
+		.test_id = "dut",
+		.test_tick_f = test_tick,
+		.test_main_f = entrypoint_dut,
+	},
+	{
+		.test_id = "peer",
+		.test_tick_f = test_tick,
+		.test_main_f = entrypoint_peer,
+	},
+	BSTEST_END_MARKER,
+};
```

Comment:
Could the `test_pre_init_f` or `bst_test_post_init_t` function be used to setup the backchannels?

Reply:
I thought about it for a while, but I didn't want to be too generic.